protocol/protocol.go

```diff
@@ -236,7 +238,7 @@ func CalculateChainNonce(prevReply, blind []byte) (nonce [NonceSize]byte) {
 // reply), the blind (needed to prove correct chaining to an external party)
 // and the request itself.
 func CreateRequest(rand io.Reader, prevReply []byte) (nonce, blind [NonceSize]byte, request []byte, err error) {
-	versionBytes := []byte{0x02, 0x00, 0x00, 0x80}
+	versionBytes := []byte{0x03, 0x00, 0x00, 0x80}
```

Comment:
Consider use binary,

```suggestion 
binary.LittleEndian.PutUint32(versionBytes,Version)
```